### PR TITLE
Make autocyber tutorial runnable on publicly-released dataset

### PR DESCRIPTION
**Pull request by aht**
The tutorial dataset is now released to s3://h1st-tutorial-autocyber/

It is based on https://zenodo.org/record/3267184#.XpHta1NKhQJ with important processing done by Arimo. The reason is that recreating realistic message frequency for each CAN ID is crucial for this problem. Simply following along the tutorial would help you understand why this is the case.

I will update the https://h1st.ai/ site content after merge.

Solves #8

### CODE REVIEW

**Review by nqbao — COMMENTED** [review]
can we organize `aegis_datagen.py` using class(es), with some documentation about the generation logic? Right now it's too hard for me to follow.

**Comment by nqbao** [comment]
Can @nhanitvn @TgithubJ or @TheVinhLuong102 review the data science part?

File: examples/AutomotiveCybersecurity/util.py [hunk 1]
```diff
@@ -19,30 +19,23 @@ def gen_windows(df, window_size, step_size):
 
 def load_data(num_files=None, shuffle=False):
     fs = s3fs.S3FileSystem(anon=False)
-    train_normal_files = ['s3://' + f for f in fs.glob(config.AUTOCYBER_DATA_PATH + "/train/Normal/*.csv", recursive=True)]
-    train_attack_files = ['s3://' + x for x in fs.glob(config.AUTOCYBER_DATA_PATH + "/train/Attack/*/*.csv", recursive=True)]
-    test_normal_files = ['s3://' + x for x in fs.glob(config.AUTOCYBER_DATA_PATH + "/test/public/Normal/*.csv", recursive=True)]
-    test_attack_files = ['s3://' + x for x in fs.glob(config.AUTOCYBER_DATA_PATH + "/test/public/Attack/*/*.csv", recursive=True)]
+    AUTOCYBER_DATA_PATH = "s3://h1st-tutorial-autocyber"
```
Comment (nqbao) [inline examples/AutomotiveCybersecurity/util.py:22]: How's about downloading as a zip file, then extracting and glob on local disk? Then we can remove the s3 dependency

File: tests/e2e/test_notebook.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    def test_e2e_autocyber_tutorial(self):
+        try:
+            subprocess.check_call(['ipython', '-V'])
+        except:
+            raise SkipTest('IPython is not installed')
+
+        nb_file = os.path.join(os.path.dirname(__file__), "AutoCyberTutorial.ipynb")
+        nb_folder = "examples/AutomotiveCybersecurity/notebooks"
+
+        shutil.copy2(nb_file, nb_folder)
```
Comment (nqbao) [inline tests/e2e/test_notebook.py:18]: we can use `tempfile.TemporaryDirectory` if you want to create a directory to be clean by the end.

File: examples/AutomotiveCybersecurity/util.py (hunk 1 above)
Comment (ctn) [reply examples/AutomotiveCybersecurity/util.py:22]: Agreeing that requiring s3 availability/access is too user-unfriendly.

**Comment by TgithubJ** [comment]
3. Using H1st.AI to Encode Human Insights as a Model and Harmonize Human + ML in a H1st.Graph
To give an example, each second might contain 100 of CAN messages per car. If we have a fleet with just 1000 cars, each driven 1h per day, then a FPR of 0.00001 at message-level still means that each day we have 0.00001 x 100msg x 3600s x 1000cars = 3600 false positive events per day!

=> Need to multiply 24 hours. 
=> I think we need to remove 1000cars because IDS is installed on each car.

**Comment by TgithubJ** [comment]
import h1st as h1

SENSORS = ["SteeringAngle", "CarSpeed", "YawRate", "Gx", "Gy"]

class MsgFreqEventDetectorModel(h1.Model):  
    def load_data(self, num_files=None):  
        return util.load_data(num_files). 

=> Should include `import util`

**Comment by TgithubJ** [comment]
The nice things about h1st.Model that we can easily save/load them. By default, the “model”, “stats” and “metrics” properties are persisted and they support a variety of flavors & data structure.
m.persist()

=> @aht I think it is better to save a model with proper name ? so that users can understand how to use it. Auto-generated version can make a confusion.

**Comment by TgithubJ** [comment]
results.keys()
dict_keys(['window_starts', 'event_detection_results'])

=> @aht It will be better if we can provide some explanation on the result. As a user, s/he can wonder why we see two keys.

**Comment by TgithubJ** [comment]
And we should see that starting we can detect attacks evemts.

=> @aht Minor typo. evemts -> events
=> And we should remove 'starting' from this sentence ?

**Comment by TgithubJ** [comment]
@aht Each python code script don't have consistent import style. Sometimes import h1st and sometimes don't. And it looks like we also need to import util.py ?

**Comment by TgithubJ** [comment]
@aht I think we updated our folder structure. No more models/ folder I think ?

**Comment by TgithubJ** [comment]
Event-level confusion matrix
[[7218    0]
 [  18 1012]]
Event TPR = 0.9825, FPR = 0.0000

=> @aht, i think it is good to show the Message-level result as well because our graph's final prediction is message-level and it is good to provide the comparison of msg-level detection performance between ML approach and H1st approach. (to make it apple-to-apple comparison).

**Comment by ctn** [comment]
@TgithubJ can you go ahead and make the changes? @aht is going to be unavailable, and we need this PR merged.

I'll review your changes.

**Comment by TgithubJ** [comment]
Understood. Let me handle it.

**Comment by TgithubJ** [comment]
@aht maybe it is better to use images in local img folder rather than getting them from docs.arimo.com to make it self-contained.

**Comment by aht** [comment]
I've made all the requested changes:

- [x] Using zip file over HTTP instead of S3 for data download
- [x] Documentation of data source & changes,
- [x] Flat directory structure
- [x] Editing
- [x] Make sure the E2E test case passes

**Comment by aht** [comment]
I've put a staging version here: http://docs.arimo.com/H1ST_AI_Tutorial/Introduction.html.

Please review.

**Review by nqbao — APPROVED** [review]
lgtm. Let's wait for the UT to pass then we can merge and improve more later.

**Pull request merged**